go/logic/applier.go

```diff
@@ -1001,15 +1002,19 @@ func (this *Applier) ApplyDMLEventQuery(dmlEvent *binlog.BinlogDMLEvent) error {
 			if err != nil {
 				return err
 			}
+			rollback := func(err error) error {
+				tx.Rollback()
+				return err
+			}
```

Comment:
@shlomi-noach just make sure "A transaction must end with a call to Commit or Rollback." and consistent with ApplyDMLEventQueries